### DISCUSSION TITLE
iOS App Zone Share Button Functionality

### DISCUSSION THREAD

**Discussion by brandongalbraith** [electricitymaps/electricitymaps-contrib#4216] [Ideas]
When using the ElectricityMap iOS app, it would be helpful to be able to share a link to a zone from the zone being viewed. In the below iOS screenshot, I included an annotation to where the share icon might be considered for placement. cc @madsnedergaard per conversation in Slack @ https://electricitymap-cntrib.slack.com/archives/CLEER5FQQ/p1652914891822299

![image](https://user-images.githubusercontent.com/350893/171720887-e5a33ef2-273d-444f-b46d-e65188966f74.png)

**Comment by VIKTORVAV99** [comment]
I think this would be a small but nice improvement. 👍🏻 

I'm not totally sure on the placement but then again there aren't many options that would work in a good way.

This should be pretty straight forward to implement by creating a new button component that implements [Navigator.canShare()](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/canShare) and [Navigator.share()](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) to first check if a platform is supported or sharing is allowed and show a share button if it is.

This method wouldn't support all platforms right now but it would use the native share menu of the operating system and if a platform added support in the future it would then automatically be be supported.

**Comment by madsnedergaard** [comment]
Thanks for bringing it up here :)

I think that would be really cool, regarding placement I'd probably put it at the bottom next to the Slack and Twitter buttons. Putting it at the top will be too much IMO in an already cluttered interface, especially for zones with long names that are estimated (US).

Another thing that comes to mind is that when clicking a share button, what do you expect is shared - is it normally "just" a link? :)

**Reply by VIKTORVAV99** [reply]
But the Twitter and Slack buttons are not shown on mobile. 👀

If we use Navigator.share() the relevant things to share would be a title, url and text comprised as a single share object. It would be up to the target to render these so it might differ based on how it's shared.

Example from https://mdn.github.io/dom-examples/web-share/ when shared via email on Windows with the following properties:
```js
let shareData = {
        title: 'MDN',
        text: 'Learn web development on MDN!',
        url: 'https://developer.mozilla.org',
      }
```
![image](https://user-images.githubusercontent.com/30777521/171846142-666ceac5-1b5e-414f-aac0-17ccb3c816a4.png)
